src/analysis/ml.py

```diff
@@ -289,3 +287,22 @@ def hac_horizontal(dataframe: pd.DataFrame, output_png: str, output_file: str, l
     clusters_df.to_csv(output_file, sep='\t', index=False)
     make_required_dirs(output_png)
     plt.savefig(output_png, bbox_inches="tight", dpi=DPI)
+
+def ensemble_network(dataframe: pd.DataFrame, output_file:str):
+    """
+    this method will sum up the binary values in the provided dataframe
+    -  we're counting the number of times an edge appears in a set of pathway reconstruction networks
+    edges that appear more frequently will have a higher number, thus are more likey to be robust,
+    so this information can be used to include an edge to include in a final network
+    @param datafram: binary dataframe of edge comparison between algorithms from summarize_networks
+    @param output_file: the file name to save the ensemble network
+    """
+    # Compute row sums, then dropped the index
+    row_sums = np.sum(dataframe, axis=1).reset_index()
+    # Rename the columns
+    row_sums.columns = ['Edges', 'Frequency']
+
+    make_required_dirs(output_file)
+    #TODO: user won't be able to open this up in excel  (as well as some of the other files in this code)
+    # should we care about that, or should I change everything to comma seperated and save as csv
```

Comment:
We could try to match the pathway output file format for the individual pathway graphs. For example
```
B	A	1
B	C	1
```